ros2param/ros2param/verb/set.py

```diff
@@ -37,14 +37,29 @@ def add_arguments(self, parser, cli_name):  # noqa: D102
             'node_name', help='Name of the ROS node')
         arg.completer = NodeNameCompleter(
             include_hidden_nodes_key='include_hidden_nodes')
-        parser.add_argument(
-            '--include-hidden-nodes', action='store_true',
-            help='Consider hidden nodes as well')
+
         arg = parser.add_argument(
-            'parameter_name', help='Name of the parameter')
+            'parameters', nargs='*',
+            help='List of parameter name and value pairs i.e. "int_param 1 str_param hello_world"')
         arg.completer = ParameterNameCompleter()
+
         parser.add_argument(
-            'value', help='Value of the parameter')
+            '--include-hidden-nodes', action='store_true',
+            help='Consider hidden nodes as well')
+
+    def build_parameters(self, params):
+        parameters = []
+        if len(params) % 2:
+            raise RuntimeError('Must pass list of parameter name and value pairs')
```

Comment:
I feel like we shouldn't raise an exception here; previously, this would result in printing out a usage message and returning with shell result 2.  I think we should try to emulate that.

Reply:
That's true -- printing out the whole backtrace and all is rather ugly. I've added an action to the new parameter argument list to maintain the same error behavior as before

https://github.com/ros2/ros2cli/blob/2e900095aa87f9581c8dda0b9fa7d619034e6756/ros2param/ros2param/verb/set.py#L32-L41

```
λ ros2 param set /add_two_ints_server
usage: ros2 param set [-h] [--spin-time SPIN_TIME] [-s]
                      [--no-daemon] [--include-hidden-nodes]
                      node_name [parameters ...]
ros2 param set: error: No parameter pairs specified
λ echo $?
2
λ ros2 param set /add_two_ints_server true
usage: ros2 param set [-h] [--spin-time SPIN_TIME] [-s]
                      [--no-daemon] [--include-hidden-nodes]
                      node_name [parameters ...]
ros2 param set: error: Must provide parameter name and value pairs
```